lighthouse-core/config/config.js

```diff
@@ -162,6 +163,22 @@ function assertValidGatherer(gathererInstance, gathererName) {
   }
 }
 
+/**
+ * Throws if pluginName is invalid or (somehow) collides with a category in the
+ * configJSON being added to.
+ * @param {LH.Config.Json} configJSON
+ * @param {string} pluginName
+ */
+function assertValidPluginName(configJSON, pluginName) {
+  if (!/^lighthouse-plugin-/.test(pluginName)) {
+    throw new Error(`plugin name '${pluginName}' does not start with 'lighthouse-plugin-'`);
+  }
+
+  if (configJSON.categories && configJSON.categories[pluginName]) {
```

Comment:
> is this effectively a no duplicates check now that we have the other restriction?

yeah, it's overkill, but if for some reason you somehow end up with a category named `lighthouse-plugin-whatever` in your config, this will prevent a plugin named `lighthouse-plugin-whatever` from being merged in.

I don't know why you'd do that, but it feels like being merged into an existing category is something that shouldn't ever be allowed, and checking it at this stage is the only way to really do it (since it's just a regular `LH.Config.Json` after parsing).